sw/device/lib/testing/otbn_testutils.c

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/otbn_testutils.h"
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/otbn.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * Otbn context to be used by the isr handler, it should be assigned with the
+ * current otbn context pointer before enabling the irq.
+ */
+static struct isr_context *otbn_isr_obj;
+
+/**
+ * Provides external IRQ handling for otbn tests.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * It performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Compute the OTBN peripheral.
+ * 3. Compute the otbn irq.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ */
+void ottf_external_isr(void) {
```

Comment:
Sorry I should have been more clear earlier, this should go in the test src file itself, i.e., in `sw/device/tests/otbn_irq_test.c`. The reason is the same as before, i.e., testutils libs are designed to be shared beteween tests, and if I have another test that wants to use the OTBN testutils lib, but also wants to override the default `ottf_external_isr` symbol, I'm going to run into issues. This means you will have to move the `isr_context` struct to the test src file as well, but that shouldn't be an issue since you pass it to the other testutils functions below.

Reply:
I got it and I think that splitting the ISR service in between the testutils and the test seems not ideal. So I decided to make this PR as simple as possible and propose a solution for an ISR service to address the code duplication among the tests in a future PR (Maybe using the chain of responsibility pattern).

Reply:
It looks like you fixed this, so marking as resolved. Thanks!